mbtiles/src/mbtiles.rs

```diff
@@ -223,7 +277,94 @@ impl Mbtiles {
             self.check_agg_tiles_hashes(&mut conn).await
         }
     }
-
+    pub async fn statistics<T>(&self, conn: &mut T) -> MbtResult<Statistics>
+    where
+        for<'e> &'e mut T: SqliteExecutor<'e>,
+    {
+        let file_size =
+            PathBuf::from(&self.filepath).metadata().unwrap().len() as f64 / 1024.0 / 1024.0;
+        let page_size = query!("PRAGMA page_size;")
+            .fetch_one(&mut *conn)
+            .await?
+            .page_size;
+        let tile_infos_query = query!(
+            r#"SELECT
+                zoom_level AS zoom,
+                count( ) AS count,
+                min( length( tile_data ) ) / 1024.0 AS smallest,
+                max( length( tile_data ) ) / 1024.0 AS largest,
+                avg( length( tile_data ) ) / 1024.0 AS average,
+                min(tile_column) as min_tile_x,
+                min(tile_row) as min_tile_y,
+                max(tile_column) as max_tile_x,
+                max(tile_row) as max_tile_y 
+            FROM tiles
+            GROUP BY zoom_level"#
+        );
+        let mbt_type = self.detect_type(&mut *conn).await?;
+        let level_rows = tile_infos_query.fetch_all(&mut *conn).await?;
+        let mut level_details: Vec<LevelDetail> = level_rows
+            .into_iter()
+            .map(|r| {
+                let zoom = r.zoom.unwrap() as u8;
+                let count = r.count as u32;
+                let tile_length = 40075016.7 / (2_u32.pow(zoom as u32)) as f64;
+
+                let smallest = r.smallest.unwrap_or(0.0) as f32;
+                let largest = r.largest.unwrap_or(0.0) as f32;
+                let average = r.average.unwrap_or(0.0) as f32;
+
+                let min_tile_x = r.min_tile_x.unwrap();
+                let min_tile_y = r.min_tile_y.unwrap();
+                let max_tile_x = r.max_tile_x.unwrap();
+                let max_tile_y = r.max_tile_y.unwrap();
+
+                let minx = -20037508.34 + min_tile_x as f64 * tile_length;
```

Comment:
bbox should be in longitude/latitude coordinates (-180..180 for longitude, approximately -86..+86 for latitude). You will need to do a proper conversion